_posts/2020-05-03-dynamic-ipv6-playground.md

```diff
@@ -0,0 +1,121 @@
+---                            
+title: "Dynamic IPv6 Playground"
+date: 2020-05-03               
+categories:                    
+  - blog                       
+tags:                          
+  - ipv6                       
+---                            
+                               
+In this post we will take a tour into the dynamic IPv6 setup, examining the
+varios options and scenarios it presents.
+
+This post is a direct continuation to the
+[Static IPv6 Playground](../static-ipv6-playground) post.
+The tooling and the basic setup will remain the same.
+
+## Basic setup
+Please revisit the setup description in the
+[static IPv6 basic setup](../static-ipv6-playground/#basic-setup) section.
+
+## The Server
+An IPv6 dynamic setup consist of a Router Advertisement (RA) service and an
+optional DHCPv6 service which serves an entire network.
+
+### Router Advertisement
+In an IPv6 network, routers send RA messages that include the needed
+information for setting up a host interface
+[IPv6 configuration](https://tools.ietf.org/html/rfc4862).
+The data included in the RA message includes the network address
+(with its prefix length), routing information and potentially other extended
+network information (e.g. [DNS](https://tools.ietf.org/html/rfc8106) entries).
+
+### DHCPv6
+Although RA is sufficient for setting up a client host interface with IPv6
+configuration, DHCPv6 complements it with additional information in
+scenarios where additional options are in need.
+The latest [RFC](https://tools.ietf.org/html/rfc8415) even proposes a full
+alternative to the stateless address autoconfiguration (SLAAC), but is probably
+less common in the industry.
+
+DHCPv6 mainly complements SLAAC by allowing a specific IPv6 address to be set
```

Comment:
Not necessarily, there is also `ra-stateless` which uses DHCPv6, too.
Maybe "Stateful DHCPv6 complements SLAAC ..." would be more precise?

Reply:
Maybe "Stateful DHCPv6 complements RA ..."  is better than "Stateful DHCPv6 complements SLAAC ..." ?

DHCPv6 can be used in a stateless way, too. In the stateless way, the client manages it's own address.

According to my understanding there are two relevant bits in the RA:

- Managed address configuration, controls if the host generates it's own address
- Other configuration, controls if the host tries to get more information via DHCPv6

There are the useful combinations:

- 00: only RA, called SLAAC, 
  - client managed IPv6 address
  - RA contains prefix length and dns server
- 01: RA, and no client specific DHCPv6 information, called "stateless DHCP"
  - client managed IPv6 address,
  - additional info like NTP server via DHCPv6
- 11: RA, and client specific information in DHCP, called "stateful DHCP"
  -  server managed IP address,

But this might not fit to the RFEs, I just noticed this in wireshark.
Information in DHCPv6 might overlap with information from RA. If they are not consistent,  different clients might behave in a different way.

Reply:
I think you summary is correct and worth adding to be more precise. I will use your suggested fix in the text and add the options as an appendix for those who may look for more details.
Thanks!